sshtunnel.py

```diff
@@ -512,165 +521,148 @@ def __init__(
         Or use `forwarder.local_bind_port` for getting local forwarding port if
         you use only one tunnel.
         """
-        if logger:
-            self.logger = logger
-            # Ensure paramiko.transport has a console handler
-            check_paramiko_handlers(logger=logger)
-        else:
-            self.logger = create_logger()
+        self.logger = logger or create_logger()
+        # Ensure paramiko.transport has a console handler
+        check_paramiko_handlers(logger=logger)
+
+        self.ssh_address_or_host = ssh_address_or_host
```

Comment:
It is not necessary to keep trying to keep all arguments state. In this case, it slightly increases the complexity of the solutions. The smaller number of private arguments and methods the better (not only for python).